frontend/src/store/summaryDeclaration.js

```diff
@@ -241,23 +204,29 @@ export const useSummaryDeclarationStore = defineStore('summaryDeclaration', {
       }
       try {
         this.setIsMainLoading(true);
-        //get application ID from the appMap so the page doesn't break when viewing historical CR records.
-        let payload = (await ApiService.apiAxios.get(`${ApiRoutes.APPLICATION_SUMMARY}/${appID}`)).data;
+
+        // get application ID from the appMap so the page doesn't break when viewing historical CR records.
+        let applicationSummaryResponse = await ApiService.apiAxios.get(`${ApiRoutes.APPLICATION_SUMMARY}/${appID}`);
+        const payload = applicationSummaryResponse.data;
+
         let summaryModel = {
           organization: undefined,
           application: payload.application,
-          facilities: payload.facilities,
           ecewe: undefined,
         };
 
-        summaryModel.facilities = summaryModel.facilities?.filter((fac) => {
-          return navBarStore.navBarList?.findIndex((item) => item.facilityId === fac.facilityId) > -1;
-        });
+        const facilities = payload.facilities
+          .filter((fac) => {
+            return navBarStore.navBarList?.findIndex((item) => item.facilityId === fac.facilityId) > -1;
+          })
+          .map(mapFacility);
+
+        this.facilities = facilities;
 
         this.setSummaryModel(summaryModel);
         this.setIsMainLoading(false);
 
-        let isSummaryLoading = new Array(summaryModel.facilities.length).fill(true);
+        let isSummaryLoading = true;
```

Comment:
do we need to define isSummaryLoading if we are just using it to set the store? I see on line 251 we are setting it with a Boolean value instead.

Reply:
Ew, moreover, the expected datatype of `isSummaryLoading` is an array of facilities :/ - it's a weird data point for incremental facility loading. I've got a commit for this.